### ISSUE TITLE
 feature: check for options (-e etc) in shebang

### ISSUE THREAD

**Issue by pabs3** [koalaman/shellcheck#1248]
#### For new checks and feature suggestions

- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet that shellcheck should flag:

```sh

#!/bin/sh -e
false
echo true
```

#### Here's a snippet that illustrates the problem:

```
$ ./foo.sh 
$ sh ./foo.sh 
true
$ sh -x ./foo.sh 
+ false
+ echo true
true
```

#### Here's what shellcheck currently says:

```
$ shellcheck foo.sh
```

#### Here's what I wanted or expected to see:

```
$ shellcheck foo.sh
In foo.sh line 1:
#!/bin/sh -e
          ^-- SCXXXX: Options in shebang lines are lost when passing the script to sh, use 'set -e' instead.
```

**Comment by austin987** [comment]
+1

I stumbled upon this while investigating build issues with PrawnOS (see https://github.com/SolidHal/PrawnOS/pull/154). If shellcheck had found this, would've saved me a lot of debugging time :)
